### PR TITLE
Write shader log in separate directory

### PR DESCRIPTION
**Pull request by raven02**
Respond to the comment from @tambry  https://github.com/RPCS3/rpcs3/issues/1631#issuecomment-207745854

> Slightly unrelated, but it would be nice to instead create the VertexProgram and FragmentProgram files in a separate folder. It's quite annoying to have the bin folder filled up with 100's of them.

### CODE REVIEW

**Comment by raven02** [comment]
Done and rebased.

**Comment by raven02** [comment]
Done and rebased.

**Comment by raven02** [comment]
Done and rebased.

File: rpcs3/Emu/RSX/D3D12/D3D12PipelineState.h [hunk 1]
```diff
@@ -130,7 +130,7 @@ struct D3D12Traits
 			}
 		}
 
-		fs::file(fs::get_config_dir() + "FragmentProgram" + std::to_string(ID) + ".hlsl", fs::rewrite).write(shader);
+		fs::file(fs::get_config_dir() + "shaderlog/" + "FragmentProgram" + std::to_string(ID) + ".hlsl", fs::rewrite).write(shader);
```
Comment (Nekotekina) [inline rpcs3/Emu/RSX/D3D12/D3D12PipelineState.h]: Two string literals can be merged into one (at all positions)

File: rpcs3/rpcs3.cpp [hunk 2]
```diff
@@ -96,7 +96,7 @@ cfg::map_entry<std::function<std::shared_ptr<GSRender>()>> g_cfg_gs_render(cfg::
 	{ "Null", PURE_EXPR(std::make_shared<NullGSRender>()) },
 	{ "OpenGL", PURE_EXPR(std::make_shared<GLGSRender>()) },
 #ifdef _MSC_VER
-	{ "DX12", PURE_EXPR(std::make_shared<D3D12GSRender>()) },
+	{ "Direct3D 12", PURE_EXPR(std::make_shared<D3D12GSRender>()) },
```
Comment (Nekotekina) [inline rpcs3/rpcs3.cpp]: This is too long, maybe D3D12?

Comment (raven02) [inline rpcs3/rpcs3.cpp]: Sure.

**Comment by raven02** [comment]
Done.

**Pull request merged**